### ISSUE TITLE
The plugin can't be built with Gradle 7.0

### ISSUE THREAD

**Issue by mateuszkwiecinski** [Karumi/Shot#218]
### Expected behaviour
I expect I can build the plugin with recently released Gradle 7.0

### Actual behaviour
When invoked with Gradle 6.8.3 the plugin shows a warning:
```
The maven plugin has been deprecated. This is scheduled to be removed in Gradle 7.0. Please use the maven-publish plugin instead. Consult the upgrading guide for further information: https://docs.gradle.org/6.8.3/userguide/upgrading_version_5.html#legacy_publication_system_is_deprecated_and_replaced_with_the_publish_plugins
        at release_9utnj6klpoi6ezucc9mhrk9tn.run((...)/Shot/release.gradle:1)
        (Run with --stacktrace to get the full stack trace of this deprecation warning.)
```
When invoked withi Gradle 7.0 the plugin fails

### Steps to reproduce
1. Update Gradle to 7.0
2. run `./gradlew`

### Version of the library
`100477e492b839c8a74372a2aec46b068b713ca4`

I decided to report this as a new issue as most probably it'll require additional attention to make sure the plugin publishing still works.

**Comment by mateuszkwiecinski** [comment]
😞 
This is actually blocked by: https://github.com/maiflai/gradle-scalatest/issues/92 as the build fails with
```
Caused by: java.lang.NullPointerException: gradleUserHome
        at org.gradle.process.internal.DefaultExecActionFactory.root(DefaultExecActionFactory.java:96)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at com.github.maiflai.ScalaTestAction.makeAction(ScalaTestAction.groovy:70)
        at com.github.maiflai.ScalaTestAction$makeAction.callStatic(Unknown Source)
        at com.github.maiflai.ScalaTestAction.execute(ScalaTestAction.groovy:35)
        at com.github.maiflai.ScalaTestAction.execute(ScalaTestAction.groovy)
```

btw, have you considered migrating plugin to Kotlin maybe? 😛 I see there are mixed Groovy and Scala and Kotlin sources - I searched what could be the reason of such setup, but I didn't find any.

**Comment by kasabov** [comment]
@mateuszkwiecinski, make sure your `--gradle-user-home` exists.
